src/test/java/org/folio/circulation/domain/OverdueFineCalculatorServiceTest.java

```diff
@@ -283,6 +285,82 @@ public void shouldCreateFeeFineTypeWhenFeeFineTypeDoesNotExist()
     verify(accountRepository, times(1)).create(any());
   }
 
+  @Test
+  public void shouldNotCreateFeeFineRecordWhenOverduePolicyDoesNotExist()
+    throws ExecutionException, InterruptedException {
+
+    final Loan loan = createLoan().withOverdueFinePolicy(OverdueFinePolicy.unknown(null));
+
+    when(overdueFinePolicyRepository.findOverdueFinePolicyForLoan(any()))
+      .thenReturn(completedFuture(succeeded(loan)));
+
+    CheckInProcessRecords records = new CheckInProcessRecords(
+      CheckInByBarcodeRequest.from(createCheckInByBarcodeRequest()).value())
+      .withLoan(loan);
+
+    overdueFineCalculatorService.createOverdueFineIfNecessary(records, CHECKIN).get();
+
+    verifyNoInteractions(feeFineRepository);
+    verifyNoInteractions(itemRepository);
+    verifyNoInteractions(feeFineOwnerRepository);
+    verifyNoInteractions(accountRepository);
+  }
+
+  @Test
+  public void shouldNotCreateFeeFineRecordWhenLoanIsNull()
+    throws ExecutionException, InterruptedException {
+
+    CheckInProcessRecords records = new CheckInProcessRecords(
+      CheckInByBarcodeRequest.from(createCheckInByBarcodeRequest()).value())
+      .withLoan(null);
+
+    overdueFineCalculatorService.createOverdueFineIfNecessary(records, CHECKIN).get();
+
+    verifyNoInteractions(feeFineRepository);
+    verifyNoInteractions(overdueFinePolicyRepository);
+    verifyNoInteractions(itemRepository);
+    verifyNoInteractions(feeFineOwnerRepository);
+    verifyNoInteractions(accountRepository);
+  }
+
+  @Test
+  public void shouldNotCreateFeeFineWhenLoanIsNotOverdue()
+    throws ExecutionException, InterruptedException {
+
+    DateTime dueDateInFuture = DateTime.now(DateTimeZone.UTC).plusDays(1);
+    final Loan loan = createLoan().changeDueDate(dueDateInFuture);
+
+    CheckInProcessRecords records = new CheckInProcessRecords(
+      CheckInByBarcodeRequest.from(createCheckInByBarcodeRequest()).value())
+      .withLoan(loan);
+
+    overdueFineCalculatorService.createOverdueFineIfNecessary(records, CHECKIN).get();
+
+    verifyNoInteractions(feeFineRepository);
+    verifyNoInteractions(overdueFinePolicyRepository);
+    verifyNoInteractions(itemRepository);
+    verifyNoInteractions(feeFineOwnerRepository);
+    verifyNoInteractions(accountRepository);
+  }
+
+  @Test
+  public void shouldNotCreateFeeFineForRenewalWhenShouldForgiveOverdueFine()
+    throws ExecutionException, InterruptedException {
+
+    final Loan loan = createLoan(); // OverdueFinePolicy.forgiveOverdueFine is true
```

Comment:
If `forgiveOverdueFine` being true is important to describe the behaviour of this test, please state that explicitly in the test setup rather than in a comment.

Reply:
Done